### ISSUE TITLE
Gapcursor and focus don't play nicely together (fix in comment)

### ISSUE THREAD

**Issue by carlobeltrame** [ueberdosis/tiptap#1899]
**Description**
When using `editor.commands.focus` to focus a Gapcursor position, the focus behaves as though the Gapcursor plugin weren't installed.

**Steps to reproduce the bug**
1. Have an editor with Gapcursor and some block elements
2. Use `editor.commands.focus(position)` to focus a position that should result in a Gapcursor (e.g. position 0 in the provided CodeSandbox)
3. Focusing and typing text works, but no gapcursor ever appears, which is confusing for the users

**CodeSandbox**
I created a CodeSandbox to help you debug the issue:

* Vue: https://codesandbox.io/s/tiptap-issue-template-forked-etu0y?file=/src/components/Tiptap.vue

**Expected behavior**
If the Gapcursor plugin is active, whenever a position is focused, Tiptap should determine whether a Gapcursor or a TextSelection makes more sense at that position.
In other words, if you follow the steps to reproduce the bug, and then press the left arrow key, that's the state I'd expect right after calling `editor.commands.focus()`.

**Environment?**
 - operating system: Ubuntu
 - browser: Firefox
 - mobile/desktop: Desktop
 - tiptap version: 2.0.0-beta96

**Additional context**
Prosemirror has to make the same decision in this method already: https://github.com/ProseMirror/prosemirror-gapcursor/blob/master/src/index.js#L56. That function is not exported unfortunately, but according to https://github.com/ProseMirror/prosemirror/issues/1201 we should be able to simply use `new GapCursor($resolvedPos)` when the gapcursor plugin is installed. Let me know if you'd like me to prepare a PR for this.

**Comment by carlobeltrame** [comment]
According to my local tests, we should be able to overwrite the `focus` command in the GapCursor extension, and copy all the code from https://github.com/ueberdosis/tiptap/blob/main/packages/core/src/commands/focus.ts and replace [this line](https://github.com/ueberdosis/tiptap/blob/main/packages/core/src/commands/focus.ts#L84) with this:
```js
const resolvedPos = editor.state.doc.resolve(resolvedFrom)
const selection = (resolvedFrom === resolvedEnd) && GapCursor.valid(resolvedPos) ?
  new GapCursor(resolvedPos) :
  TextSelection.create(doc, resolvedFrom, resolvedEnd)
```

When reproducing this in the CodeSandbox, I also noticed a bug in [resolveSelection](https://github.com/ueberdosis/tiptap/blob/main/packages/core/src/commands/focus.ts#L8) in the same file: The conditional should be an explicit `null` check, or else it will return `null` when passed a position of `0`.

When I fix that as well, I can reproduce my fix working in this modified CodeSandbox: https://codesandbox.io/s/tiptap-issue-template-forked-etu0y?file=/src/components/Tiptap.vue

**Comment by carlobeltrame** [comment]
As a workaround, I was able to create this small extension that fixes the issue:
```js
import {GapCursor} from 'prosemirror-gapcursor'
import {Extension} from '@tiptap/core'

const GapCursorFocus = Extension.create({
  onFocus({ editor }) {
    editor.commands.focusWithGapcursor()
  },

  addCommands() {
    return {
      focusWithGapcursor: () => ({ editor, tr }) => {
        const { from, to } = editor.state.selection
        if (from !== to) {
          return true
        }
        const resolvedPos = editor.state.doc.resolve(from)
        if (GapCursor.valid(resolvedPos)) {
          tr.setSelection(new GapCursor(resolvedPos))
        }
      }
    }
  }
})

export default GapCursorFocus
```

But ideally, this should be fixed in the GapCursor extension itself.

**Comment by carlobeltrame** [comment]
Sorry stalebot, this problem didn't magically disappear.

**Comment by carlobeltrame** [comment]
@hanspagel Please reopen this, it's still a problem.

**Comment by sipec** [comment]
Another workaround: after `focus()` do [createParagraphNear()](https://tiptap.dev/api/commands/create-paragraph-near) which will add a new empty paragraph node and focus it if the currently focused node is a block node (like an image). This only solves some use cases however.

**Comment by carlobeltrame** [comment]
@sipec I created a new issue because stalebot does not understand the difference between issues which are unanswered by the issue authors and issues which are unanswered by the maintainers. #3074
